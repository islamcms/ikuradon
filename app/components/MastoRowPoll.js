import React, { memo } from "react";
import { View, StyleSheet, Text } from "react-native";
import PropTypes from "prop-types";
import {open as openUrl} from "../util/url";
import t from "../services/I18n";

function MastoRowPoll({poll}){
    let {id, expired, expires_at,voters_count,votes_count,voted,own_votes,options,multiple,emojis} = poll;
    //console.log({id, expired, expires_at,voters_count,votes_count,voted,own_votes,options,multiple,emojis});
    return (
        <View style={[]}>
            {
                options.map((v)=>{
                    return <Text>{v.title+": "+v.votes_count+t("polls.votes")}</Text>
                })
            }
        </View>
    );
}

export default MastoRowPoll;