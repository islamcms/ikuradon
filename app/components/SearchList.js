import React, { useContext } from "react";
import { View, FlatList, StyleSheet, TouchableOpacity } from "react-native";
import { ThemeContext } from "react-native-elements";
import { ListItem } from "react-native-elements";
import * as searchConst from "../constants/search";

export default function SearchList({ type, data }){
    const { theme }= useContext(ThemeContext);
    return (
        <View style={styles.container}>
            <FlatList
                keyExtractor={data => type === searchConst.TYPE_HASHTAGS ? data.name : data.id}
                data={data}
                renderItem={({ item }) =>{
                    switch (type){
                        case searchConst.TYPE_ACCOUNTS:
                            return <ListItem
                                leftAvatar={{ source: { uri: item.avatar } }}
                                title={item.display_name}
                                subtitle={item.acct}
                                bottomDivider
                                onPress={() => null}
                            />;
                        case searchConst.TYPE_STATUSES:
                            return null;
                        case searchConst.TYPE_HASHTAGS:
                            return <ListItem
                                title={"#" + item.name}
                                subtitle={item.history[0] && item.history[0].accounts}
                                bottomDivider
                                onPress={() => null}
                            />;
                    }
                }}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container:{
        flex: 1
    },
});