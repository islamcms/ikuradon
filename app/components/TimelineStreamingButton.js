import React, { useState, useContext, useRef, useEffect } from "react";
import { View, StyleSheet, TouchableOpacity } from "react-native";
import { FontAwesome } from "@expo/vector-icons";
import { useSelector, useDispatch } from "react-redux";

import { ThemeContext } from "react-native-elements";
import { start, stop, receive } from "../actions/actioncreators/streaming";
import { STREAMING } from "../constants/api";

const CurrentUserReducerSelector = state => state.currentUserReducer;

export default function TimelineStreamingButton({ type }){
    const dispatch = useDispatch();
    const { instance, access_token } = useSelector(CurrentUserReducerSelector);
    const { theme } = useContext(ThemeContext);
    const [enabled, useEnabled] = useState(false);
    const webSocket = useRef(null);
    const streamSwitch = () => {
        useEnabled(!enabled);
    };
    useEffect(() => {
        if (enabled && instance){
            // ON
            let stream;
            switch (type) {
                case "federal":
                    stream = "public";
                    break;
                case "local":
                    stream = "public:local";
                    break;
                case "home":
                default:
                    stream = "user";
                    break;
            }
            let url = instance.urls.streaming_api + STREAMING.url + "?access_token=" + access_token + "&stream=" + stream;
            webSocket.current = new WebSocket(url);
            webSocket.current.onopen = () => {
                // connection opened
                console.log("[WS] OPEN:" + type);
            };
            webSocket.current.onmessage = (message) => {
                let data = JSON.parse(message.data);
                if (data !== null){
                    let { event, payload } = data;
                    dispatch(receive(type, event, JSON.parse(payload)));
                }
            };
            webSocket.current.onclose = (event) => {
                console.log("[WS] CLOSE:" + type + " event:" + JSON.stringify(event));
            };
            webSocket.current.onerror = (event) => {
                console.log("[WS] ERROR:" + type + " event:" + JSON.stringify(event));
                dispatch(stop(type));
                useEnabled(false);
            };
            dispatch(start(type));
        } else {
            dispatch(stop(type));
            if (webSocket.current !== null && webSocket.current.readyState === WebSocket.OPEN){
                webSocket.current.close();
            }
        }
        return;
    }, [enabled]);
    useEffect(() => {
        return () => {
            // unmount
            dispatch(stop(type));
            if (webSocket.current !== null && webSocket.current.readyState === WebSocket.OPEN){
                webSocket.current.close();
                console.log("[WS] UNMOUNTCLOSE:" + type);
            }
        };
    }, []);
    return (
        <View>
            <TouchableOpacity onPress={() => streamSwitch()} style={styles.view}>
                <FontAwesome name="feed" size={24} color={enabled === true ? theme.customColors.primaryComplementary : theme.customColors.primaryBackground} />
            </TouchableOpacity>
        </View>
    );
}
const styles = StyleSheet.create({
    view: {
        paddingTop: 4,
        paddingBottom: 4,
        paddingLeft: 12,
        paddingRight: 4
    }
});