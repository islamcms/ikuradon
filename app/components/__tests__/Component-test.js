import React from "react";
import renderer from "react-test-renderer";

import theme from "../../themes/default";
import { ThemeContext } from "react-native-elements";

import NotificationsList from "../NotificationsList";
import NotificationsRow from "../NotificationsRow";
import TimelineTootButton from "../TimelineTootButton";
import TootButton from "../TootButton";
import TootImageClip from "../TootImageClip";
import UserList from "../UserList";
import VisibilityModal from "../VisibilityModal";
import TimelineStreamingButton from "../TimelineStreamingButton";
import TimelineLeftHeader from "../TimelineLeftHeader";
import TimelineCenterHeader from "../TimelineCenterHeader";
import Search from "../Search";
import SearchList from "../SearchList";

import * as searchConst from "../../constants/search";
import MastoRowPoll from "../MastoRowPoll";

jest.mock("@expo/vector-icons/build/FontAwesome", () => "FontAwesome");
jest.mock("react-native-elements/src/icons/Icon", () => "Icon");

const mockCurrent = {
    user_credentials: {
        id: "1",
        acct: "example",
        display_name: "example",
        avatar: "http://example.com/example.png",
    },
    domain: "example.com",
    access_token: "ACCEESS_TOKEN",
    notification_count: 0,
    instance: {
        contact_account:{},
        description: "description"
    }
};

const mockPoll = {
    id: "1",
    expires_at: "",
    expired: false,
    multiple: false,
    votes_count: 0,
    voters_count: 0,
    voted: false,
    options: [],
};


it("<MastoRowPoll />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <MastoRowPoll poll={mockPoll} />
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});

//it("<NotificationsList />", () => {
//    const result = renderer.create(
//        <ThemeContext.Provider value={{ theme }}>
//            <NotificationsList type={"notifications"} />
//        </ThemeContext.Provider>
//    );
//    expect(result).toMatchSnapshot();
//});

//it("<NotificationsRow />", () => {
//    const result = renderer.create(
//        <ThemeContext.Provider value={{ theme }}>
//            <NotificationsRow item={{}} current={current} actions={actions} background={config.backgroundImage !== null} />
//        </ThemeContext.Provider>
//    );
//    expect(result).toMatchSnapshot();
//});

it("<Search />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <Search />
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});

it("<SearchList />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <SearchList type={searchConst.TYPE_ACCOUNTS} data={[]} />
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});

it("<TimelineCenterHeader />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <TimelineCenterHeader fixedTitle={false} onPress={() => null} current={mockCurrent}/>
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});

it("<TimelineLeftHeader />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <TimelineLeftHeader isBack={true} onPress={() => null} />
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});

//it("<TimelineStreamingButton />", () => {
//    const result = renderer.create(
//        <ThemeContext.Provider value={{ theme }}>
//            <TimelineStreamingButton 
//                type={"home"}
//            />
//        </ThemeContext.Provider>
//    );
//    expect(result).toMatchSnapshot();
//});

it("<TimelineTootButton />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <TimelineTootButton 
                onPress={() => null}
                enabled={true}
                loading={false}
            />
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});

it("<TootButton />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <TootButton onPress={()=>null} />
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});

it("<TootImageClip />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <TootImageClip callbackMediaAttachments={()=>null} />
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});

//it("<UserList />", () => {
//    const result = renderer.create(
//        <ThemeContext.Provider value={{ theme }}>
//            <UserList current={mockCurrent} onSelect={()=>null} />
//        </ThemeContext.Provider>
//    );
//    expect(result).toMatchSnapshot();
//});

it("<VisibilityModal />", () => {
    const result = renderer.create(
        <ThemeContext.Provider value={{ theme }}>
            <VisibilityModal onSelect={()=>null} />
        </ThemeContext.Provider>
    );
    expect(result).toMatchSnapshot();
});