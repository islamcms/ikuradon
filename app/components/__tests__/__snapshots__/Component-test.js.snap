// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MastoRowPoll /> 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginLeft": 10,
      "marginRight": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#657786",
        },
        Object {
          "flex": 1,
          "fontSize": 14,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 5,
        },
      ]
    }
  >
    0人
    (0票)
     - 
    NaN時間
  </Text>
</View>
`;

exports[`<Search /> 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "borderBottomColor": "#e1e1e1",
        "borderBottomWidth": 1,
        "borderTopColor": "#e1e1e1",
        "borderTopWidth": 1,
        "padding": 8,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderBottomWidth": 0,
            "borderColor": undefined,
            "borderRadius": 3,
            "flexDirection": "row",
            "minHeight": 30,
            "overflow": "hidden",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <Icon
            name="search"
            size={18}
            type="material"
          />
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Search"
          platform="default"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": undefined,
              "flex": 1,
              "fontSize": 18,
              "marginLeft": 10,
              "minHeight": 40,
            }
          }
          testID="searchInput"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginRight": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": undefined,
            "fontSize": 12,
            "height": 0,
            "margin": 0,
            "padding": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e3e3e3",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 5,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "borderRightColor": undefined,
          "borderRightWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF3300",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "grey0": "#657786",
                  "grey1": "#2b90d9",
                  "grey2": "#8899a6",
                  "platform": Object {
                    "android": Object {},
                    "ios": Object {},
                  },
                  "primary": "#FF3300",
                },
                "customColors": Object {
                  "char": "#000000",
                  "charBackground": "#FFFFFF",
                  "charReverse": "#FFFFFF",
                  "item": Object {
                    "bookmark": "#4c6ef5",
                    "boost": "#2b90d9",
                    "favourite": "#ffd27d",
                    "none": "#8899a6",
                  },
                  "link": "#1b95e0",
                  "primaryBackground": "#FFFFFF",
                  "primaryComplementary": "#00CCFF",
                  "tootButton": "#FF5500",
                },
              }
            }
          >
            Accounts
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRightColor": undefined,
          "borderRightWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#8899a6",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "grey0": "#657786",
                  "grey1": "#2b90d9",
                  "grey2": "#8899a6",
                  "platform": Object {
                    "android": Object {},
                    "ios": Object {},
                  },
                  "primary": "#FF3300",
                },
                "customColors": Object {
                  "char": "#000000",
                  "charBackground": "#FFFFFF",
                  "charReverse": "#FFFFFF",
                  "item": Object {
                    "bookmark": "#4c6ef5",
                    "boost": "#2b90d9",
                    "favourite": "#ffd27d",
                    "none": "#8899a6",
                  },
                  "link": "#1b95e0",
                  "primaryBackground": "#FFFFFF",
                  "primaryComplementary": "#00CCFF",
                  "tootButton": "#FF5500",
                },
              }
            }
          >
            Toots
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#8899a6",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "grey0": "#657786",
                  "grey1": "#2b90d9",
                  "grey2": "#8899a6",
                  "platform": Object {
                    "android": Object {},
                    "ios": Object {},
                  },
                  "primary": "#FF3300",
                },
                "customColors": Object {
                  "char": "#000000",
                  "charBackground": "#FFFFFF",
                  "charReverse": "#FFFFFF",
                  "item": Object {
                    "bookmark": "#4c6ef5",
                    "boost": "#2b90d9",
                    "favourite": "#ffd27d",
                    "none": "#8899a6",
                  },
                  "link": "#1b95e0",
                  "primaryBackground": "#FFFFFF",
                  "primaryComplementary": "#00CCFF",
                  "tootButton": "#FF5500",
                },
              }
            }
          >
            HashTags
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<SearchList /> 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<TimelineCenterHeader /> 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 4,
        "paddingTop": 4,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://example.com/example.png",
        }
      }
      style={
        Object {
          "borderRadius": 15,
          "height": 30,
          "marginRight": 5,
          "width": 30,
        }
      }
    />
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontWeight": "bold",
          "height": 30,
          "width": 200,
        }
      }
    >
      @
      example.com
    </Text>
  </View>
</View>
`;

exports[`<TimelineLeftHeader /> 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingBottom": 4,
        "paddingLeft": 12,
        "paddingRight": 4,
        "paddingTop": 4,
      }
    }
  >
    <FontAwesome
      color="#FFFFFF"
      name="chevron-left"
      size={20}
    />
  </View>
</View>
`;

exports[`<TimelineTootButton /> 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingBottom": 4,
        "paddingLeft": 12,
        "paddingRight": 4,
        "paddingTop": 4,
      }
    }
  >
    <FontAwesome
      color="#FFFFFF"
      name="pencil"
      size={24}
    />
  </View>
</View>
`;

exports[`<TootButton /> 1`] = `
<View
  style={
    Object {
      "elevation": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <Icon
    color="#FF5500"
    name="pencil"
    onPress={[Function]}
    raised={true}
    reverse={true}
    size={28}
    type="font-awesome"
  />
</View>
`;

exports[`<TootImageClip /> 1`] = `
<RCTScrollView
  horizontal={true}
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": undefined,
          },
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#657786",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 5,
            "marginRight": 5,
            "opacity": 1,
            "width": 70,
          }
        }
      >
        <FontAwesome
          color="#2b90d9"
          name="plus-circle"
          size={26}
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<VisibilityModal /> 1`] = `
<View
  style={
    Object {
      "height": 260,
      "width": 300,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": undefined,
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <FontAwesome
        color="#2b90d9"
        name="globe"
        size={26}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          public
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 15,
            }
          }
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          Post to public timelines
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": undefined,
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <FontAwesome
        color="#2b90d9"
        name="unlock"
        size={26}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          unlisted
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 15,
            }
          }
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          Do not post to public timelines
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": undefined,
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <FontAwesome
        color="#2b90d9"
        name="lock"
        size={26}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          private
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 15,
            }
          }
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          Post to followers only
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": undefined,
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <FontAwesome
        color="#2b90d9"
        name="envelope"
        size={26}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          direct
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 15,
            }
          }
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          Post to mentioned users only
        </Text>
      </View>
    </View>
  </View>
</View>
`;
